experiments/snippets/csvUpload/CSV_Parser.xml

```diff
@@ -0,0 +1,232 @@
+						<field id="Picklist" uniqueid="fi-35BB-19158"/>
+						<field id="Text" uniqueid="fi-35BB-19159"/>
+						<field id="TextArea" uniqueid="fi-35BB-19160"/>
+						<field id="URL" uniqueid="fi-35BB-19161"/>
+					</fields>
+					<filtering enableSearch="false"/>
+					<actions/>
+					<rowActions/>
+					<massActions/>
+					<exportProperties useTableColumns="true"/>
+					<sorting enable="false"/>
+				</skuid__table>
+			</components>
+			<styles>
+				<spacing top="4" bottom="4" left="4" right="4"/>
+			</styles>
+			<background/>
+		</skuid__wrapper>
+	</components>
+	<resources>
+		<javascript>
+			<jsitem location="inlinesnippet" name="parseCsvData" cachelocation="false">var $ = skuid.$;
+var inputData = skuid.$M("Input").getFirstRow().Data;
+var reject = function(err) {
+    console.error(err);
+    return Promise.reject(err);
+};
+
+if (!inputData) {
+    return reject("No CSV data provided.");
+}
+
+var result = Papa.parse(inputData);
+
+if (result.errors &amp;&amp; result.errors.length) {
+    return reject("Unable to parse CSV File: " + result.errors.join(", "));
+}
+
+if (!result.data.length) {
+    return reject("No data found in CSV file");
+}
+
+var dataModel = skuid.$M("Data");
+var rows = result.data;
+// Fuzzy-match the field names to data
+var columnNames = rows[0];
+var fieldNamesByColumnIndex = [];
+var modelFieldIdsByLabel = {};
+skuid.$.each(dataModel.fields, function(i, field) {
+   modelFieldIdsByLabel[field.label.toLowerCase()] = field.id; 
+});
+var mappedAllColumns = true;
+skuid.$.each(columnNames, function(columnIndex, columnName) {
+    var useFieldName;
+    // Ideal case -- column name matches field name exactly
+    if (dataModel.getField(columnName)) {
+        useFieldName = columnName;
+    } else {
+        // Try lower-cased column name label
+        var lowerCaseColumnName = columnName.toLowerCase();
+        if (modelFieldIdsByLabel[lowerCaseColumnName]) {
+            useFieldName = modelFieldIdsByLabel[lowerCaseColumnName];
+        }
+    }
+    if (!useFieldName) {
+        useFieldName = -1;
+    }
+    fieldNamesByColumnIndex.push(useFieldName);
+});
+
+// Now parse the data and create rows
+$.each(rows, function(i, sourceRow) {
+
+    // Skip first row, which is the column names
+    if (i === 0) return true;
+
+    dataModel.createRow({
+        doAppend: true,
+        additionalConditions: $.map(sourceRow, function(columnValue, columnIndex) {
+            var useFieldName = fieldNamesByColumnIndex[columnIndex];
+            if (useFieldName !== -1) {
+                return {
+                    field: useFieldName,
+                    value: columnValue,
+                };   
+            } else {
+                return null;
+            }
+        }),
+    });
+});
+</jsitem>
+			<jsitem location="staticresource" name="papaparse.min.js" cachelocation="false" url="" content_type="text/javascript">var params = arguments[0],
```

Comment:
I tried to upload PapaParse5 as a static resource and Salesforce got mad since it's above 10 MB in size.

To avoid that (and get this page running faster with platform-agnostic instructions), could we use a CDN link instead? With this as an external resource the page still worked for me 👍 
```suggestion
			<jsitem location="external" name="papaparse.min.js" cachelocation="false" url="https://cdnjs.cloudflare.com/ajax/libs/PapaParse/5.3.1/papaparse.min.js" content_type="text/javascript">var params = arguments[0],
```

Reply:
If we go with this, we can remove those extra instructions from the README